libwebauthn/src/transport/virt/device.rs

```diff
@@ -0,0 +1,239 @@
+// Extracted (and slightly modified) from the fido-authenticator crate:
+//   https://github.com/Nitrokey/fido-authenticator/blob/main/tests/virt/mod.rs
+//
+// License: Apache-2.0 or MIT
+//
+// Authors:
+// - Robin Krahl <robin@nitrokey.com>
+
+use super::pipe::Pipe;
+use std::{
+    borrow::Cow,
+    cell::RefCell,
+    fmt::Debug,
+    path::Path,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread,
+    time::{Duration, SystemTime},
+};
+
+use ctaphid::{
+    error::{RequestError, ResponseError},
+    HidDevice, HidDeviceInfo,
+};
+use ctaphid_dispatch::{Channel, Dispatch, Requester, DEFAULT_MESSAGE_SIZE};
+use fido_authenticator::{Authenticator, Config, Conforming};
+use littlefs2::{object_safe::DynFilesystem, path, path::PathBuf};
+use rand::{
+    distributions::{Distribution, Uniform},
+    RngCore as _,
+};
+use tracing::info;
+use trussed::{
+    backend::BackendId,
+    platform::Platform as _,
+    store::Store as _,
+    virt::{self, StorageConfig, StoreConfig},
+};
+use trussed_staging::virt::{BackendIds, Client, Dispatcher};
+
+// see: https://github.com/Nitrokey/nitrokey-3-firmware/tree/main/utils/test-certificates/fido
+const ATTESTATION_CERT: &[u8] = include_bytes!("../../../../data/fido-cert.der");
```

Comment:
Should this be within `libwebauthn`? Not sure if the crates.io build would succeed otherwise.

Ideally, can we use the cargo environment variable here?

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates